scripts/mvftoms.py

```diff
@@ -143,76 +147,119 @@ def main():
                      'bpcal': 'CALIBRATE_BANDPASS,CALIBRATE_FLUX',
                      'target': 'TARGET'}
 
-    usage = "%prog [options] <dataset> [<dataset2>]*"
+    def casa_style_int_list(val, argparse=False, opt_unit="m"):
+        """ returns list of ints """
+        RangeException = ArgumentTypeError if argparse else ValueError
+        if val.strip() == "" or val.strip() == "*":
+            return None
+        elif re.match(r"^(\d+)(~\d+[" + opt_unit +
```

Comment:
It seems like a bit of a waste to do a full regexp and then parse the string using split, filter, replace, etc.

How about simplifying both approaches by splitting the string on commas first and then doing the following regexp?

```Python
int_or_range = r"^(\d+)(~(\d+)[{}]?)?$".format(opt_unit)
match = re.match(int_or_range, val)
if not match:
    raise RangeException(...)
if match.group(3):
    # range
    rmin = int(match.group(1))
    rmax = int(match.group(3))
    vals.append(list(range(rmin, rmax + 1))
else:
    # single int
    vals.append(int(match.group(1))
```

Reply:
neater... will change it